### PR TITLE
feat(cvrp): generator

### PR DESCRIPTION
**Pull request by surana01**
Implement a `Generator` class for the CVRP environment which is responsible for generating the CVRP instances. This will include the following additions:
- A `generator.py` script added to the `cvrp` folder which implements an abstract class `Generator` and a `RandomGenerator` to create a random uniform CVRP instance. 
- A `generator_test.py` script to test the generator. 
- Modifications to `env.py` to include the generator in the `init` method and changes to `conftest.py` to include a `DummyGenerator` for testing purposes.

### CODE REVIEW

File: jumanji/environments/routing/cvrp/generator.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+class UniformGenerator(Generator):
+    """Instance generator that generates a random uniform instance of the capacitated vehicle
+    routing problem. Given the number of nodes, maximum capacity of the vehicle and maximum demand
+    of the nodes, the generation works as follows. The coordinates of the cities and the depot are
+    randomly sampled from a uniform distribution on the unit square. The demands of the nodes are
+    randomly sampled integers from the interval [1, max_demand], and the demand of the depot is set
+    to 0.
+    """
+
+    def __init__(self, num_nodes: int, max_capacity: int, max_demand: int):
```
Comment (clement-bonnet) [inline jumanji/environments/routing/cvrp/generator.py:68]: Could we add a docstring to the init?

File: jumanji/environments/routing/cvrp/env.py [hunk 2]
```diff
@@ -96,33 +97,36 @@ class CVRP(Environment[State]):
...
     ):
         """Instantiates a `CVRP` environment.
 
         Args:
-            num_nodes: number of city nodes in the environment. Defaults to 20.
-            max_capacity: maximum capacity of the vehicle. Defaults to 30.
-            max_demand: maximum demand of each node. Defaults to 10.
+            generator: `Generator` whose `__call__` instantiates an environment instance.
+                Default option is 'UniformGenerator' which defaults to a CVRP instance with 20
+                cities, a maximum vehicle capacity of 30, and a maximum city demand of 10.
```
Comment (clement-bonnet) [inline jumanji/environments/routing/cvrp/env.py:109]:
```suggestion
                The default option is 'UniformGenerator' which randomly generates
                CVRP instances with 20 cities sampled from a uniform distribution.
```

**Review by clement-bonnet — APPROVED** [review]
Looks almost perfect, thanks!

File: jumanji/environments/routing/cvrp/generator.py (hunk 1 above)
Comment (surana01) [reply jumanji/environments/routing/cvrp/generator.py:68]: Done!

File: jumanji/environments/routing/cvrp/env.py (hunk 2 above)
Comment (clement-bonnet) [reply jumanji/environments/routing/cvrp/env.py:109]: Right, we probably want that.

Comment (clement-bonnet) [reply jumanji/environments/routing/cvrp/env.py:109]: What about the following instead?
```suggestion
                The default option is 'UniformGenerator' which randomly generates
                CVRP instances with 20 cities sampled from a uniform distribution,
                a maximum vehicle capacity of 30, and a maximum city demand of 10.
```

**Review by clement-bonnet — APPROVED** [review]

**Pull request merged**